### PR TITLE
Feature/new vehicle data turn signal

### LINKED ISSUE #1859: [SDL 0107] New vehicle data - TurnSignal
Proposal: [New vehicle data - TurnSignal](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0107-New-vehicle-data-turnSignal.md)

> This proposal is about adding a new vehicle data to get the value of the turn signal.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/328

### PR DESCRIPTION
**Pull request by EKuliiev**
Fixes #1859

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit tests modified according to new parameter.
TODO: Provide ATF tests.

### Summary
Add new vehicle data parameter 'turnSignal' to the MOBILE/HMI apis and testing files.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by EKuliiev** [comment]
This PR may considered to be ready after ATF provided and 'PASSED'.

**Comment by EKuliiev** [comment]
@LuxoftAKutsan , improvement task "Multiple duplications of *preloaded*.json configurations in application_manager tests" created

File: src/components/interfaces/MOBILE_API.xml [hunk 1]
```diff
@@ -4056,6 +4073,9 @@
         <param name="instantFuelConsumption" type="Boolean" mandatory="false">
             <description>The instantaneous fuel consumption in microlitres</description>
         </param>
+        <param name="turnSignal" type="Boolean" mandatory="false">
+            <description>See TurnSignal</description>
+         </param>
```
Comment (AByzhynar) [inline src/components/interfaces/MOBILE_API.xml]: @EKuliiev Please align lines

**Comment by EKuliiev** [comment]
implemented:
https://github.com/smartdevicelink/sdl_core/pull/2293

**Comment by LuxoftAKutsan** [comment]
@ekuliev @vkushnirenko-luxoft  please delete redundant branches after closing implemented feature PRs

**Comment by EKuliiev** [comment]
@LuxoftAKutsan done

**Pull request closed without merging**